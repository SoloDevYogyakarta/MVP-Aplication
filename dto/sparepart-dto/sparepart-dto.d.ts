export declare class SparepartField {
    text: string;
    id?: number;
}
