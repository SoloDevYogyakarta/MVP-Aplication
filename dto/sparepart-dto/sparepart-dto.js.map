{"version":3,"file":"sparepart-dto.js","sourceRoot":"","sources":["../../../src/dto/sparepart-dto/sparepart-dto.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,qDAAiE;AAEjE,MAAa,cAAc;CAO1B;AAPD,wCAOC;AALC;IADC,IAAA,0BAAQ,GAAE;;4CACE;AAIb;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;0CACD"}