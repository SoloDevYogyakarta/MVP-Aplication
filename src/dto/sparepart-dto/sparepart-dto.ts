import { IsNumber, IsOptional, IsString } from 'class-validator';

export class SparepartField {
  @IsString()
  text: string;

  @IsNumber()
  @IsOptional()
  id?: number;
}
