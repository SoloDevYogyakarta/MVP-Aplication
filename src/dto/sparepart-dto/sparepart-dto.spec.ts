import { faker } from '@faker-js/faker';
import { validate } from 'class-validator';
import { SparepartField } from './sparepart-dto';

describe('SparepartField', () => {
  it('should to be defined', () => expect(SparepartField).toBeDefined());

  it('render correctly', () => expect(SparepartField).toMatchSnapshot());

  it('should to be successfully', async () => {
    const data = new SparepartField();
    data.text = faker.lorem.paragraph();
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });
});
