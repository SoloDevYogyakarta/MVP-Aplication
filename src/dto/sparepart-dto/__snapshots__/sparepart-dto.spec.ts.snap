// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SparepartField render correctly 1`] = `[Function]`;
